pkg/kubelet/secret/secret_manager.go

```diff
@@ -209,8 +209,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 2 minutes (which is a lot, but...)
+	ttl := 2 * time.Minute
```

Comment:
kubelet should not be making decisions based on the cluster-wide knowledge. It should not even be aware of other nodes in the cluster. If I had to pick, I'd rather add a flag...